From: SeongJae Park <sj@kernel.org>
Date: Tue, 25 Feb 2025 12:30:23 -0800
Subject: [PATCH] mm/damon/core: (fixup) set sample_interval again after
 kdamond_tune_intervals()

kdamond_tune_intervals() could updated sampling intervals.  Set
sample_interval again to reflect the change.

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/core.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/mm/damon/core.c b/mm/damon/core.c
index fa6153d95d8d..d4fc2c907cfc 100644
--- a/mm/damon/core.c
+++ b/mm/damon/core.c
@@ -2456,6 +2456,9 @@ static int kdamond_fn(void *data)
 					ctx->attrs.aggr_samples *
 					ctx->attrs.intervals_goal.aggrs;
 				kdamond_tune_intervals(ctx);
+				sample_interval = ctx->attrs.sample_interval ?
+					ctx->attrs.sample_interval : 1;
+
 			}
 			ctx->next_aggregation_sis = next_aggregation_sis +
 				ctx->attrs.aggr_interval / sample_interval;
-- 
2.39.5

